### ISSUE TITLE
Build fails against Debian Bullseye [testing]

### ISSUE THREAD

**Issue by IndelicateArgot** [zcash/zcash#4090]
### Describe the issue
I attempted to build zcash in a docker container from Debian Bullseye [testing], and the build failed. 

Here is the Dockerfile:
```Dockerfile
FROM     debian:testing
RUN      apt-get update -y && \
apt-get dist-upgrade -y && \
apt-get install -y build-essential pkg-config libc6-dev m4 g++-multilib \
autoconf libtool ncurses-dev unzip git python python-zmq \
zlib1g-dev wget curl bsdmainutils automake curl && \
rm -rf /var/lib/apt/lists/*
RUN      git clone https://github.com/zcash/zcash.git
RUN      cd zcash && ./zcutil/build.sh -j$(nproc)
```

Here is a link to the output log:
https://gist.github.com/IndelicateArgot/2b98e7f0d13e75e1ad61543bb353ee4d

### Machine specs:
```bash
uname -a
Linux [redacted] 4.15.0-55-generic #60~16.04.2-Ubuntu SMP Thu Jul 4 09:03:09 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```
*4GB RAM

Based off the following from the output log, it seems like `ar` is possibly being invoked with the wrong options:

```bash
grep -n -5 Usage: [dump text file]
10689-ar: creating libsnark.a
10690-if [ -n "" ]; then mkdir -p tmp-ar; cd tmp-ar; for AR_LIB in ; do ar x $AR_LIB; done; ar qc libsnark.a tmp-ar/*; cd ..; rm -r tmp-ar; fi;
10691-ar s libsnark.a
10692-make[3]: Leaving directory '/zcash/src/snark'
10693--s: invalid option -- 'n'
10694:Usage: ar [emulation options] [-]{dmpqrstx}[abcDfilMNoOPsSTuvV] [--plugin <name>] [member-name] [count] archive-file file...
10695-       ar -M [<mri-script]
10696- commands:
10697-  d            - delete file(s) from the archive
10698-  m[ab]        - move file(s) in the archive
10699-  p            - print file(s) found in the archive
```

**Comment by daira** [comment]
This is likely to be fixed by removing libsnark (#4060).

**Comment by Eirik0** [comment]
It's also possible (but maybe unlikely in this case) that this is related to the amount of RAM. The zcashd [user guide](https://zcash.readthedocs.io/en/latest/rtd_pages/user_guide.html) says that the minimum requirement is 5GB.

Do you mind running `ar --version` and telling the results?

**Comment by Eirik0** [comment]
@IndelicateArgot #4060 has been merged and is included in the v2.1.0 release. Has this resolved the problem for you?

**Comment by zancas** [comment]
Oh dear...   I am trying to build old versions of zcashd to test for the introduction of a bug, and I am hitting a build issue...  I _think_ it's this one.

**Comment by zancas** [comment]
> @IndelicateArgot #4060 has been merged and is included in the v2.1.0 release. Has this resolved the problem for you?

I don't think it's a bug in `ar` I think the Makefile expects the `$(AR_LIBS)` make target to evaluate to something other than an empty string...  and it doesn't.

**Comment by zancas** [comment]
This is sufficient to reproduce the bug:

`cd src/snark && ar s libsnark.a`

The above is run in a context where `libsnark.a` is present.

**Comment by zancas** [comment]
I'll try this fix:  https://github.com/HandBrake/HandBrake/issues/2272#issuecomment-524606944

**Comment by zancas** [comment]
@daira I tried the above fix and it worked for me.

**Comment by daira** [comment]
This is caused by https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=941921 I think, which is apparently fixed in binutils 2.33-2. That version is currently only packaged for Debian unstable (testing has 22.3-1), so the above workaround is probably easier.

In any case, libsnark has been removed so this is no longer an issue for current zcashd. Closing.
